### DISCUSSION TITLE
Add-on categories

### DISCUSSION THREAD

**Discussion by J-N-K** [openhab/openhab-distro#1623] [openHAB Maintainers Discussions]
I just came across a question that I could not answer myself. According to the [documentation](https://www.openhab.org/docs/developer/addons/) we have the following add-on types: `automation`, `binding`, `misc`, `persistence`, `transform`, `ui` and `voice`. In openhab-addons we use the following bundle-prefixes: `org.openhab.automation`, `org.openhab.binding`, `org.openhab.io`, `org.openhab.persistence`, `org.openhab.transform`, `org.openhab.voice` (and `org.openhab.ui` in openhab-webui). Except `misc` there is a match in a repository, but in the repository we have `io`.

I am currently developing a new add-on which is a `CurrencyProvider`, which is not a binding (no things) and would fit in the `misc` category. But `org.openhab.io` feels wrong, because it contains add-ons that integrate other eco-systems like HomeKit or Alexa. 

My question is: should we introduce `org.openhab.misc` and eventually re-factor the existing `org.openhab.io` add-ons to that?

**Comment by kaikreuzer** [comment]
I would actually tend to say that the referenced documentation is not correct; Instead of `misc`, the category really is `io` (as a technical package name), which is then called "System Integrations" on https://www.openhab.org/addons/. This is the category for all add-ons that expose openHAB in a certain way to allow integration into other systems.

I'd therefore say that your request is rather to define a new add-on category as you rightly say that a currency provider does not really fit into any of the existing ones.
The question is whether or not we could make it fit somewhere. Does it get exchange rates from an external service? Then we could probably have an "account bridge" and maybe even think of some things that might provide interesting information; the currency provider would then be yet another service that is provided, once the account bridge is set up.
In case there is no external dependency at all, I would even consider the currency provider to be a core service that be easily provided by default.

Wdyt?

**Comment by J-N-K** [comment]
Well, we also use the same categories (including `misc` instead of `io` for the [addon.xsd](https://github.com/openhab/openhab-core/blob/7d1dcd7937ac8810f64691445179b2951d0246bf/bundles/org.openhab.core.addon/schema/addon-1.0.0.xsd#L41-L50), the additional [add-on services](https://github.com/openhab/openhab-core/blob/7d1dcd7937ac8810f64691445179b2951d0246bf/bundles/org.openhab.core.addon.marketplace/src/main/java/org/openhab/core/addon/marketplace/internal/community/CommunityBundleAddonHandler.java#L45-L46), the [Main UI](https://github.com/openhab/openhab-webui/blob/b175d1b6bd0bc78621a600a2a319f79780cfa99c/bundles/org.openhab.ui/web/src/pages/addons/addons-store.vue#L116-L144) and some other places. 

This has been at least the case since https://github.com/openhab/openhab-core/pull/2508, it was named `io` before and changed because UI and forum used `misc` instead. It would be a bit difficult to now change that back again.

**Comment by kaikreuzer** [comment]
Hm, ok, so despite https://github.com/openhab/openhab-core/pull/2508 claiming to fix inconsistencies, we still have quite some inconsistencies...

But to come to a solution for your problem at hand: What do you think about my reasoning above for the currency provider?

**Comment by J-N-K** [comment]
Yes, a simple thing would be possible with (extensible) channels for the current exchange rate between two currencies (both configuration options of the channel). The required API key could be a binding-configuration parameter (used for the service), because it makes no sense to have two different API keys for retrieving the exact same data.

**Comment by andrewfg** [comment]
> Then we could probably have an "account bridge" and maybe even think of some things that might provide interesting information

We already have such bridges, things and channels for astronomical events, weather, and solar forecasts. IMHO an exchange rate provider is no different from those.
